src/systems/detachable_joint/DetachableJoint.cc

```diff
@@ -234,8 +234,75 @@ void DetachableJoint::PreUpdate(
     }
     else
     {
-      modelEntity = _ecm.EntityByComponents(
-          components::Model(), components::Name(this->childModelName));
+      auto entitiesMatchingName = entitiesFromScopedName(
+        this->childModelName, _ecm);
+
+      // TODO(arjoc): There is probably a more efficient way of combining entitiesFromScopedName
+      // With filtering.
+      // Filter for entities with only models
+      std::vector<Entity> candidateEntities;
+      for (const auto entity: entitiesMatchingName)
+      {
+        if (_ecm.EntityHasComponentType(entity, components::Model::typeId))
+        {
+          candidateEntities.push_back(entity);
+        }
+        else
+        {
+          ignwarn << "Found entity " << scopedName(entity, _ecm)
+            << " but it is not a model." << std::endl;
+        }
+      }
```

Comment:
```suggestion
      std::copy_if(entitiesMatchingName.begin(), entitiesMatchingName.end(),
                 std::back_inserter(candidateEntities),
                 [&_ecm](Entity e) { return _ecm.EntityHasComponentType(e, components::Model::typeId); });
```

I'd suggest to remove the log message as it can be spammy if there is another entity with the same name. We'll log in any case that the child model could not be found on line 335.